### PR TITLE
editoast: move train schedule base to schemas

### PR DESCRIPTION
**Pull request by Wadjetz**
part of https://github.com/OpenRailAssociation/osrd/issues/7048

- [ ] Maybe rename TrainScheduleBase to TrainSchedule ?
- [x] Move train schedule base to schemas
- [x] Move tests to schemas
- [x] Copy a json file exemple of train schedule for testing

### CODE REVIEW

**Review by woshilapin — APPROVED** [review]
Yeah, that the end of `editoast/src/schema/v2/mod.rs` 🎉 Thanks for the PR.

For the renaming of `TrainScheduleBase` into `TrainSchedule`, I agree with you, but let's wait for @leovalais to give an approval on that (he knows better than me).

**Review by leovalais — APPROVED** [review]
LGTM. Ping @flomonster for the renaming of `TrainScheduleBase` in `TrainSchedule` since he's the most involved with TrainScheduleV2. I'd be in favor of renaming unless there's a specific reason it was named this way.

**Comment by Wadjetz** [comment]
Hey @leovalais and @flomonster, let's merge this PR. It's blocking @hamz2a. We can make any potential modifications in another PR.

**Pull request merged**
